FlowerOrders/FlowerOrders/Screens/OrderDetails/OrderDetailsViewController.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  OrderDetailsViewController.swift
+//  FlowerOrders
+//
+//  Created by Paul Tiriteu on 06/08/2020.
+//  Copyright © 2020 Paul Tiriteu. All rights reserved.
+//
+
+import UIKit
+
+class OrderDetailsViewController: UIViewController {
+    @IBOutlet weak var idLabel: UILabel!
+    @IBOutlet weak var deliverButton: UIButton!
+    @IBOutlet weak var nameLabel: UILabel!
+    @IBOutlet weak var descriptionLabel: UILabel!
+    @IBOutlet weak var priceLabel: UILabel!
+    @IBOutlet weak var orderImageView: UIImageView!
+    
+    private let viewModel: OrderDetailsViewModel
+    
+    @IBAction func deliverButtonAction(_ sender: Any) {
+        viewModel.deliverOrder()
+    }
+    
+    init(viewModel: OrderDetailsViewModel) {
+        self.viewModel = viewModel
+        super.init(nibName: OrderDetailsViewController.className, bundle: Bundle(for: OrderDetailsViewController.self))
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureUI()
+    }
+    
+    private func configureUI() {
+        let order = viewModel.order
+        if let orderId = Int(exactly: order.uid) {
+            title = "Order ID: \(orderId)"
+        } else {
+            title = "Order details"
+        }
+        
+        if viewModel.order.sent == false {
```

Comment:
This could be configured in the viewModel so we avoid again the logic in the View layer.

Reply:
This is something I actually missed. At the line 40 I wanted to bring the order from the viewModel, so I can use it for the UI configuration. But I agree, we can create variables in the viewModel, which can return the info we need